iroh-gateway/src/core.rs

```diff
@@ -242,88 +221,90 @@ async fn get_ipfs(
 }
 
 #[tracing::instrument()]
-async fn get_ipns(
-    Extension(state): Extension<Arc<State>>,
-    Path(params): Path<HashMap<String, String>>,
-    Query(query_params): Query<GetParams>,
-    request_headers: HeaderMap,
+fn protocol_handler_redirect(
+    uri_param: String,
+    state: &State,
 ) -> Result<GatewayResponse, GatewayError> {
-    state.metrics.requests_total.inc();
-    let start_time = time::Instant::now();
-    // parse path params
-    let cid_param = params.get("cid").unwrap();
-    let cid = cid_param.clone();
-    let cpath = "".to_string();
-    let cpath = params.get("cpath").unwrap_or(&cpath);
+    let u = Url::parse(&uri_param);
+    if u.is_err() {
+        return Err(error(
+            StatusCode::BAD_REQUEST,
+            "invalid uri parameter",
+            state,
+        ));
+    }
+    let u = u.unwrap();
```

Comment:
`unwrap` is generally nice to avoid so sth like this is nicer

```rust
let u = match Url::parse() {
  Ok(u) => u,
  Err(e) => { return Err(..); }
};
```